### PR TITLE
chore(deps): update

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #170

Notes:
- `precompile-charcodes` is abandoned and don't have compatibility with babel@7, just change `charCodeAt` on number (keep perf)
- test with same names are prohibited so i added `(2)`/`(3)`/etc
- snapshots are changes, just update their

### CODE REVIEW

File: src/__tests__/__snapshots__/postcss.js.md [hunk 1]
```diff
@@ -124,16 +124,6 @@ Generated by [AVA](https://ava.li).
         |  ^␊
     `
 
-## bad whitespace instead of namespace token
```
Comment (evilebottnawi) [inline src/__tests__/__snapshots__/postcss.js.md]: Strange, WIP on this

Comment (evilebottnawi) [reply src/__tests__/__snapshots__/postcss.js.md]: We remove this test and forget update snapshot, strange what old version of `ava` doesn't exit with 1 code

**Comment by evilebottnawi** [comment]
No need major release right now, i want to fix:
https://github.com/postcss/postcss-selector-parser/issues/173
https://github.com/postcss/postcss-selector-parser/issues/174
https://github.com/postcss/postcss-selector-parser/issues/175

maybe it is already fixed after updating `cssesc`, but i want to add tests to avoid future regressions

File: src/__tests__/__snapshots__/postcss.js.md (hunk 1 above)
Comment (jonathantneal) [reply src/__tests__/__snapshots__/postcss.js.md]: Are you saying it is okay to remove this test?

File: src/tokenTypes.js [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-export const plus             = `+`.charCodeAt(0);
-export const equals           = `=`.charCodeAt(0);
-export const pipe             = `|`.charCodeAt(0);
-export const greaterThan      = `>`.charCodeAt(0);
-export const space            = ` `.charCodeAt(0);
-export const singleQuote      = `'`.charCodeAt(0);
-export const doubleQuote      = `"`.charCodeAt(0);
-export const slash            = `/`.charCodeAt(0);
-export const bang             = `!`.charCodeAt(0);
+export const ampersand        = 38; // `&`.charCodeAt(0);
```
Comment (jonathantneal) [inline src/tokenTypes.js]: Why has this been replaced?

Comment (evilebottnawi) [reply src/tokenTypes.js]: @jonathantneal 
> * `precompile-charcodes` is abandoned and don't have compatibility with babel@7, just change `charCodeAt` on number (keep perf)

File: src/__tests__/__snapshots__/postcss.js.md (hunk 1 above)
Comment (evilebottnawi) [reply src/__tests__/__snapshots__/postcss.js.md]: @jonathantneal yep, we remove this tests in past, but doesn't update snapshot (i don't know why), `jest` exit with non zero code when `snapshot` contains extra snapshots but `ava` doesn't do this in old versions

**Comment by jonathantneal** [comment]
Great work, @evilebottnawi. As far as semver is concerned, is this a patch release?

**Comment by evilebottnawi** [comment]
Major after update `cssesc`, need merge this too https://github.com/postcss/postcss-selector-parser/pulls (one fix)

**Comment by jonathantneal** [comment]
All set for the next major (v6.0.0)?

**Comment by evilebottnawi** [comment]
@jonathantneal yep

**Pull request merged**